### DISCUSSION TITLE
Tabby (FOSS) Console Application

### DISCUSSION THREAD

**Discussion by iam-thehelp** [GNS3/gns3-gui#3346] [General]
Hello,

Has anyone tried using Tabby as the console for GNS3? I looked around but could not find anything related to the topic.

Any information would be helpful, thank you.

**Comment by eantowne** [comment]
Looking at the Github repo (https://github.com/Eugeny/tabby/releases) it seems to be getting fairly regular updates). The real problem I see is that I cannot seem to find any documentation that provides command line options. Passing command line parameters is pretty much necessary for it to be able to be integrated with GNS3 as a console provider.

**Comment by marius-101** [comment]
The question was: how to add tabby to gns3 as a default terminal?

**Reply by grossmj** [reply]
Does Tabby take any command line parameters?

**Reply by eantowne** [reply]
From what I can tell Tabby still does not have any CLI parameters/arguments.

@marius-101, you can add it as the default terminal by going into preferences and Console Applications tab and adding as custom. The issue is that without the ability to pass an IP address and port, as well as instantiate a new tab, Tabby will not actually connect to anything, it will just start Tabby with the default session, but not connect tot he appliance that you are expecting.

For clarification this is not an issue with GNS3, but with Tabby. If you search the issues list for Tabby on it's GitHub page, you will see MANY requests for the ability to make a connection from a CLI command with Tabby. I did not see a single response from the developers reference this. Based on that, it seems like the Tabby developers have no intention of ever supporting that capability.

**Reply by grossmj** [reply]
Apparently you can use tabby to run commands. For instance: `tabby run telnet localhost 5000`
